bluemira/neutronics/params.py

```diff
@@ -0,0 +1,309 @@
+
+from dataclasses import asdict, dataclass
+from pathlib import Path
+from typing import Literal, Tuple, Union
+
+import openmc
+
+from bluemira.base.constants import raw_uc
+from bluemira.neutronics.constants import energy_per_dt
+from bluemira.neutronics.make_materials import BlanketType
+
+
+@dataclass
+class OpenMCSimulationRuntimeParameters:
+    """Parameters used in the actual simulation
+
+    Parameters
+    ----------
+    particles:
+        Number of neutrons emitted by the plasma source per batch.
+    batches:
+        How many batches to simulate.
+    photon_transport:
+        Whether to simulate the transport of photons (i.e. gamma-rays created) or not.
+    electron_treatment:
+        The way in which OpenMC handles secondary charged particles.
+        'thick-target bremsstrahlung' or 'local energy deposition'
+        'thick-target bremsstrahlung' accounts for the energy carried away by
+        bremsstrahlung photons and deposited elsewhere, whereas 'local energy
+        deposition' assumes electrons deposit all energies locally.
+        (the latter is expected to be computationally faster.)
+    run_mode:
+        see below for details:
+        https://docs.openmc.org/en/stable/usersguide/settings.html#run-modes
+    openmc_write_summary:
+        whether openmc should write a 'summary.h5' file or not.
+    cross_section_xml:
+        Where the xml file for cross-section is stored locally.
+    """
+
+    # Parameters used outside of setup_openmc()
+    parametric_source: bool  # to use the pps_isotropic module or not.
+    particles: int  # number of particles used in the neutronics simulation
+    cross_section_xml: Union[str, Path]
+    batches: int = 2
+    photon_transport: bool = True
+    electron_treatment: Literal[
+        "ttb", "led"
+    ] = "led"  # Bremsstrahlung only matters for very thin objects
+    run_mode: str = openmc.settings.RunMode.FIXED_SOURCE.value
+    openmc_write_summary: bool = False
+    # number of particles used in the volume calculation.
+    volume_calc_particles: int = int(4e8)
+
+
+@dataclass
+class BreederTypeParameters:
+    """Dataclass to hold information about the breeder blanket material
+    and design choices.
+    """
+
+    enrichment_fraction_Li6: float
+    blanket_type: BlanketType
+
+
+@dataclass(frozen=True)
+class TokamakOperationParameters:
+    """Parameters describing how the tokamak is operated,
+    i.e. where the plasma is positioned (and therefore where the power is concentrated),
+    and what temperature the plasma is at.
+
+    Parameters
+    ----------
+    reactor_power:
+        total reactor (thermal) power when operating at 100%
+    peaking_factor:
+        (max. heat flux on fw)/(avg. heat flux on fw)
+    shaf_shift:
+        shafranov shift
+        how far (towards the outboard direction) is the centre of the plasma shifted
+        compared to the geometric center of the poloidal cross-section.
```

Comment:
I called it "TokamakOperationParameters" rather than "PlasmaNeutronSourceParameters" because our plasma neutron source [uses a couple of other variables from TokamakGeometry](https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/bf9a971c2cafb180a7af5e379c0c00c5e22da0f5/bluemira/neutronics/neutronics_axisymmetric.py#L188) to create the plasma (namely major radius, minor radius, elongation, triangularity); but TokamakGeometry was used elsewhere (i.e. in make_geometry.load_fw_points), so this is the best grouping and naming that I could come up at the time.

Although that does raise a question: I don't know if major radius, minor radius, elongation, triangularity should be classified as "PlasmaGeometry" or TokamakGeometry?

As for densities and plasma geometry: see other comment: For the current neutron source, the 1D reaction-rate profile (i.e. the reaction rate at each of the closed flux surfaces, theoretically obtained by (d-density on that surface) * (t-density on that surface), and modified slightly w.r.t. temperature on that surface) is hard-coded and simply scaled by the scalar, i.e. number of neutrons. Therefore, no plasma density is needed yet.